src/client/js/app-shell.js

```diff
@@ -29,50 +29,24 @@ const routeConfig = [
 
 function init() {
   const store = initStore(window.__INITIAL_STATE__);
-  const isAppShell = window.__INITIAL_STATE__.context.isAppShell;
-  const routes = routeConfig.map(route => {
-    return Object.assign({}, route, {
+  const promiseRoutes = routeConfig.map(route => {
+    if (matchPath(window.location.pathname, route)) {
+      return route.loadComponent()
+        .then(component => Object.assign({}, route, { component }));
+    }
+
+    return Promise.resolve(Object.assign({}, route, {
       component: Loadable({
         loading: LoadingPage,
         loader: route.loadComponent
       })
-    });
+    }));
   });
 
-  if (isAppShell) {
```

Comment:
We no longer need this because the pageLoaderHOC will also always render the same loading page for the app shell page - mean no react checksum errors.